### ISSUE TITLE
ink_detection: native full-3D inference drops configured input depth padding

### ISSUE THREAD

**Issue by make-agi-quick** [ScrollPrize/villa#1507]
## Summary

Native full-3D training and native full-3D inference do not currently apply the same `model_config.input_pad_depth_to` contract.

Training explicitly pads the BCZYX model input before every forward pass:

```python
# vesuvius/src/vesuvius/ink_detection/training/train.py
return center_pad_input_depth(
    image_BCZYX, config.ink.model.input_pad_depth_to
)
```

The shared inference wrapper also supports exactly this contract:

```python
class TargetModel(nn.Module):
    def __init__(self, model, *, input_pad_depth_to=None):
        ...
    def forward(self, image_BCZYX):
        image_BCZYX = center_pad_input_depth(
            image_BCZYX, self.input_pad_depth_to
        )
        ...
```

Flat inference passes the checkpoint-resolved value correctly:

```python
model = TargetModel(
    base_model,
    input_pad_depth_to=config.model.input_pad_depth_to,
).eval()
```

but native inference currently constructs the same wrapper without it:

```python
# infer_full3d_tifxyz.py::_load_native_model
model = TargetModel(base_model.eval()).eval()
```

So for any supported native checkpoint with `model_config.input_pad_depth_to > patch_size[0]`, training sends a center-padded depth tensor to the network while native inference sends the unpadded depth tensor.

## Minimal deterministic regression

A regression does not need a real checkpoint or GPU. Use a tiny recording module whose `forward()` stores the received BCZYX shape, wrap it as native inference does, and feed depth 3 with `input_pad_depth_to=5`.

Expected training/inference contract:

```text
input to wrapper:      [B,C,3,H,W]
input seen by network: [B,C,5,H,W]
```

Current native wrapper construction yields depth 3 instead of 5.

The existing `center_pad_input_depth` tests already establish exact padding semantics, including even/odd deficits.

## Proposed fix

```diff
-    model = TargetModel(base_model.eval()).eval()
+    model = TargetModel(
+        base_model.eval(),
+        input_pad_depth_to=config.model.input_pad_depth_to,
+    ).eval()
```

Add a native inference regression asserting that `_load_native_model` / its `TargetModel` preserves the checkpoint's configured depth-padding contract.

## Scope

This is a supported-path train/inference parity defect, but I have **not** established that currently published native checkpoints use a non-null `input_pad_depth_to`, nor am I claiming an ink-accuracy improvement. The failure becomes operational whenever such a checkpoint is used.
